### ISSUE TITLE
[Question] How to pass args to pre-recieve hook

### ISSUE THREAD

**Issue by Sykomaniac** [pre-commit/pre-commit#549]
Hi,

I'm trying to leverage some of the pre-receive hooks listed at: https://github.com/github/platform-samples/tree/master/pre-receive-hooks. They require certain info to be passed (I assume this would be done through args) such as block_branch_names.sh requires: oldrev newrev refname.

Any idea how I can use pre-commit to pass these?

Apologies if this isn't the right place to ask!

**Comment by asottile** [comment]
Unfortunately these aren't currently passed along to hooks.  pre-commit is a framework built around filenames (it takes the `oldrev` and `newrev` and computes the changed files and calls each configured hook based on those -- `refname` is ignored).

I think the only thing that _could_ be done here would be to export some environment variables that hooks could use -- though I'm not sure this will be any more useful since the list of filenames is probably the only useful information to gain from that.

**Comment by Sykomaniac** [comment]
Ah thanks for letting me know!

**Comment by Sykomaniac** [comment]
Guessing there is no chance of passing the SHA's (i.e. `oldrev` and `newrev`) at some point in the future?

**Comment by asottile** [comment]
would environment variables work?  I could add that pretty easily

**Comment by Sykomaniac** [comment]
Yeah they could do. More than happy to try it out!

**Comment by asottile** [comment]
Something like this? https://github.com/pre-commit/pre-commit/pull/550

**Comment by Sykomaniac** [comment]
Brilliant. I'll give it a try tomorrow as my laptop is at work.

**Comment by Sykomaniac** [comment]
@asottile I can't seem to get this to work. 

I've converted my hook to a python package (using yours as a base). I've replaced the main with simply:

`print (os.environ.get('PRE_COMMIT_ORIGIN'))`

As I believe python automatically passes environment variables to other child scopes. However this simply gives me `None`. 

To try out your change I just dropped the extra lines into run.py in my install dir and deleted the pyc file. I'm assuming this is OK?
